extensions/3rdparty/DynamicFunctions/DynamicFunctions.php

```diff
@@ -97,7 +97,7 @@ function rand( &$parser, $a = 0, $b = 1 )
 	 * @param Parser $parser
 	 * @return bool
 	 */
-    function skin( &$parser )
+    function skin( $parser )
     {
         global $wgUser, $wgRequest;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)